### DISCUSSION TITLE
Video player .mp4 files and video formats

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#22065] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

`Vscode` allows you to open video formats and also view them via controls, see image below:

<img width="712" alt="Screenshot 2024-12-07 alle 19 33 08" src="https://github.com/user-attachments/assets/4424d79d-197a-4530-92dc-3766d76458fa">

### Environment

```
Zed: v0.165.3 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by JosephTLyons** [comment]
I'm tagging this as `image viewer` as to avoid adding another label. If we end up having support for videos, we can create a new label then to track issues and enhancement requests specifically for that.

**Comment by notpeter** [comment]
A video player is well out-of-scope for the moment.  If you have ideas for how to implement this, including the necessary libraries and integration with gpui components feel free to discuss.

**Comment by wdphoto** [comment]
Hey! Just wanted to bring this back up. Zed's popular enough and theres enough scientific and rational data usage for seeing a video file inside a code project. I can think of a thousand use-cases where this should be on the roadmap. 

Happy to take a look at it in a fork for macOS and contribute, but I'll have to see what work has been done first.

**Reply by Angelk90** [reply]
@wdphoto : As far as I know, nothing has been done.
